packages/wdio-utils/src/test-framework/testFnWrapper.ts

```diff
@@ -83,6 +95,17 @@ export const testFrameworkFnWrapper = async function (
     const testStart = Date.now()
     try {
         result = await executeAsync.call(this, specFn, retries, specFnArgs, timeout)
+        if (
+            globalThis._wdioDynamicJasmineResult &&
+            global._wdioDynamicJasmineResult.passedExpectations.length === 0 &&
+            global._wdioDynamicJasmineResult.failedExpectations.length > 0
+        ) {
+            const jasmineErr = new Error(globalThis._wdioDynamicJasmineResult.failedExpectations[0]?.message)
+            jasmineErr.stack = globalThis._wdioDynamicJasmineResult.failedExpectations[0].stack
+            throw jasmineErr
+        } else {
+            console.error(afterHookErrMessage)
+        }
```

Comment:
So we can simplify this:
```suggestion
        if (globalThis._wdioDynamicJasmineResultError) {
            error = globalThis._wdioDynamicJasmineResultError
            globalThis._wdioDynamicJasmineResultError = undefined
        }
```

Reply:
I think it makes sense then to name the variable `_wdioDynamicJasmineResultError`

Reply:
Done